src/main/java/com/sfuentes/ticket/Ticket.java

```diff
@@ -0,0 +1,35 @@
+package com.sfuentes.ticket;
+
+import com.sfuentes.grocery.Client;
+import com.sfuentes.grocery.Grocery;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public class Ticket {
+
+  @NonNull
+  private final Grocery grocery;
+  private final Client client;
+  private final double payment;
+
+  private double vat;
+  private double totalToPay;
+  private double change;
+
+  public void calculateVAT() {
```

Comment:
If I'm correct by following this approach and looking at the tests, seems that you will create a completely new Grocery object for each sale, which I'm not sure if is the best way to do it, because a Grocery store should be handling more than one sale during the day.

Reply:
Also, seems that by doing it in this way you won't be able to calculate the total sales amount  at the end of the day as per the problem states here:
" Al final del día deberá imprimir la cantidad de dinero que debe haber en la caja."
Since you are creating a new grocery store per client you loose the accumulate of all sales.

but again, maybe there is a valid reason, let's discuss this.

Reply:
Seems that these comments have to do with the previous one, so let's have a quick chat about the design.

Reply:
I get i, the design of my program was wrong.